include/pybind11/exec.h

```diff
@@ -0,0 +1,159 @@
+    }
+    if (!local)
+        local = global;
+
+    auto st = static_cast<std::string>(string);
+    PyObject *res = PyRun_String(st.c_str() , Py_eval_input, global.ptr(), local.ptr());
+
+    if (res == nullptr)
+        throw error_already_set();
+
+    return {res, false};
+}
+
+inline object exec (str string, object global = object(), object local = object()) {
+    if (!global) {
+        if (PyObject *g = PyEval_GetGlobals())
+            global = object(g, true);
+        else
+            global = dict();
+    }
+    if (!local)
+        local = global;
+
+    auto st = static_cast<std::string>(string);
+    PyObject *res = PyRun_String(st.c_str() , Py_file_input, global.ptr(), local.ptr());
+
+    if (res == nullptr)
+        throw error_already_set();
+
+    return {res, false};
+}
+
+inline object exec_statement (str string, object global = object(), object local = object()) {
+    if (!global) {
+        if (PyObject *g = PyEval_GetGlobals())
+            global = object(g, true);
+        else
+            global = dict();
+    }
+    if (!local)
+        local = global;
+
+    auto st = static_cast<std::string>(string);
+    PyObject *res = PyRun_String(st.c_str() , Py_single_input, global.ptr(), local.ptr());
+    if (res == nullptr)
+        throw error_already_set();
+
+    return {res, false};
+}
+
+inline object exec_file(str filename, object global = object(), object local = object()) {
+    // Set suitable default values for global and local dicts.
+    if (!global) {
+    if (PyObject *g = PyEval_GetGlobals())
+        global = object(g, true);
+    else
+        global = dict();
+    }
+    if (!local) local = global;
+    // should be 'char const *' but older python versions don't use 'const' yet.
+
+    auto f = static_cast<std::string>(filename);
+
+    // Let python open the file to avoid potential binary incompatibilities.
+#if PY_VERSION_HEX >= 0x03040000
+    const static int close_it = 1;
+    FILE *fs = _Py_fopen(f.c_str(), "r");
+#elif PY_VERSION_HEX >= 0x03000000
+    const static int close_it = 1;
+    PyObject *fo = Py_BuildValue("s", f.c_str());
+    FILE *fs = _Py_fopen(fo, "r");
+    Py_DECREF(fo);
+#else
+    const static int close_it = 0;
+    PyObject *pyfile = PyFile_FromString(&f.front(), const_cast<char*>("r"));
+    if (!pyfile)
+    	throw std::invalid_argument(std::string(f) + " : no such file");
+    object file(pyfile, false);
+    FILE *fs = PyFile_AsFile(file.ptr());
+#endif
+    if (fs == nullptr)
+    	throw std::invalid_argument(std::string(f) + " : could not be opened");
+
+    PyObject* res = PyRun_FileEx(fs,
+                f.c_str(),
+                Py_file_input,
+                global.ptr(), local.ptr(),
+				close_it);
+
+    if (res == nullptr)
+        throw error_already_set();
+
+    return {res, false};
+
+}
+
+inline object exec (const std::string &string, object global = object(), object local = object()) {
+    return exec(str(string), global, local);

```

Comment:
Minor, but here you convert `string -> str -> string`, would it make sense to do it the other way around and avoid double cast?